ray-operator/controllers/raycluster_controller.go

```diff
@@ -202,6 +202,75 @@ func (r *RayClusterReconciler) reconcileServices(instance *rayiov1alpha1.RayClus
 	return nil
 }
 
+func (r *RayClusterReconciler) needRemoveOldPod(template v1.PodTemplateSpec, pod v1.Pod) bool {
+	if pod.Status.Phase == v1.PodRunning && pod.ObjectMeta.DeletionTimestamp == nil {
+		if len(template.Spec.Containers) != len(pod.Spec.Containers) {
+			return true
+		}
+		cmap := map[string]*v1.Container{}
+		for _, container := range pod.Spec.Containers {
+			cmap[container.Name] = &container
+		}
+		for _, container1 := range template.Spec.Containers {
+			if container2, ok := cmap[container1.Name]; ok {
+				if container1.Image != container2.Image {
+					log.Info("image name do not match")
+					// image name do not match
+					return true
+				}
+				if len(container1.Resources.Requests) != len(container2.Resources.Requests) {
+					// resource requests entries do not match
+					log.Info("resource requests entries do not match")
+					return true
+				}
+				if len(container1.Resources.Limits) != len(container2.Resources.Limits) {
+					// limits requests entries do not match
+					log.Info("limits requests entries do not match")
+					return true
+				}
+				for name, quantity1 := range container1.Resources.Requests {
```

Comment:
I remember there're some tools to compare the resources in kubernetes pkgs.  Let me see if I can find it

Reply:
are you talking about pkg/api/v1/resource/helpers.go?